managed/devops/bin/yb_platform_util_py3.py

```diff
@@ -0,0 +1,483 @@
+    :return: None
+    """
+    configure_json = modify_universe_config(input_file, universe_name)
+    universe_config_json = post_universe_config(configure_json, base_url, customer_uuid, auth_uuid)
+    # add the missing fields in the config json.
+    universe_config_json["clusterOperation"] = "CREATE"
+    universe_config_json["currentClusterType"] = "PRIMARY"
+    # Call universe config api to get the update config data.
+    universe_config_json = post_universe_config(
+        universe_config_json,
+        base_url,
+        customer_uuid,
+        auth_uuid
+    )
+    if universe_config_json:
+        create_universe_from_config(universe_config_json, base_url, customer_uuid, auth_uuid)
+    else:
+        data = {
+            "data": 'Unable to Configure Universe for Customer %s' % customer_uuid,
+            "status": "failed",
+            "error": 'Unable to Configure Universe for Customer %s' % customer_uuid
+        }
+        handle_response(data)
+
+
+@exception_handling
+def get_task_details(task_id, base_url, customer_uuid, auth_uuid):
+    """
+    Get details of the ongoing task.
+
+    :param task_id: Task UUID.
+    :param base_url: base url of the platform back end.
+    :param customer_uuid: authentication token for the customer.
+    :param auth_uuid: authentication token for the customer.
+    :return: None
+    """
+    task_url = base_url + "/api/v1/customers/" + customer_uuid + "/tasks/" + task_id
+    response = call_api(task_url, auth_uuid)
+    universe_json = json.loads(response.read())
+    if universe_json["status"] == "Running":
+        response = {"data": int(universe_json["percent"]), "status": "success", "error": ""}
+        print(response)
+    elif universe_json["status"] == "Success":
+        response = {"data": 100, "status": "success", "error": ""}
+        print(response)
+    elif universe_json["status"] == "Failure":
+        content = {
+            "message": "{0} failed".format(universe_json["title"]),
+            "details": universe_json["details"]
+        }
+        response = {"data": content, "status": "success",
+                    "error": content}
+        print(response)
+
+
+def create_universe_config(universe_data, universe_name):
+    """
+    Create the universe config data from the json file.
+
+    :param universe_data: Stored universe data.
+    :param universe_name: Name of the new universe to be created.
+    :return: Configured universe json.
+    """
+    configure_json = {}
+    clusters = universe_data['universeDetails']['clusters']
+    user_az_selected = universe_data['universeDetails']['userAZSelected']
+
+    excluded_keys = [
+        "uuid",
+        "storageType",
+        "awsArnString",
+        "useHostname",
+        "preferredRegion",
+        "regions",
+        "index",
+        "placementInfo"
+    ]
+
+    clusters_list = get_cluster_list(clusters, excluded_keys, universe_name)
+    configure_json["clusters"] = clusters_list
```

Comment:
Don't we need deep copy here ?

Reply:
@skorobogatydmitry we could use that but we are calling this function here only and only once in the program. also, we are not dealing with the creation of multiple universes in a single shot I believe. Until we use it for creation of single universe it would be fine I guess. But yes we could use deep copy to be on a safer side. Correct me If I am wrong.